internal/controllers/topology/cluster/desired_state.go

```diff
@@ -424,10 +424,14 @@ func (r *Reconciler) computeControlPlaneVersion(ctx context.Context, s *scope.Sc
 	}
 
 	// If the control plane is not upgrading or scaling, we can assume the control plane is stable.
-	// However, we should also check for the MachineDeployments to be stable.
-	// If the MachineDeployments are rolling out (still completing a previous upgrade), then do not pick
-	// up the desiredVersion yet. We will pick up the new version after the MachineDeployments are stable.
-	if s.Current.MachineDeployments.IsAnyRollingOut() {
```

Comment:
I wonder if we can get rid of `IsAnyRollingOut`. Doesn't seem to make sense to use it in the condition if it is not used elsewhere

Reply:
I think this also mean we don't need: RollingOut, MarkRollingOut, MarkUpgradingAndRollingOut can be only MarkUpgrading, ...

Can you check if I'm right?

Reply:
We can drop `IsAnyRollingOut` but we need to keep the rest because we use that information to check if the cluster has finished and upgrade and call the `AfterClusterUpgrade` hook accordingly.

Reply:
Hm not sure. Depends on how we define "AfterClusterUpgrade". I think it's fine to define that as all MachineDeployments have Machines with the latest version.

I'm not sure if should include rolling out as that could mean that we also wait for arbitrary MHC remediations, scale ups, unhealthiness, ... in the upgrade.

I think something like this should be okay: https://github.com/ykakarap/cluster-api/pull/62

Reply:
This way we would end up with a consistent "upgrade" definition

Reply:
@fabriziopandini Opinions?

Reply:
The current upgrading check we have should be enough to be a good indicator for the `AfterClusterUpgrade` hook. +1.